power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+
+#include "base_optimizer.hpp"
+
+#include "../auxiliary/dataset.hpp"
+#include "../common/enum.hpp"
+
+#include <boost/graph/compressed_sparse_row_graph.hpp>
+
+namespace power_grid_model::optimizer {
+namespace tap_position_optimizer {
+
+namespace detail = power_grid_model::optimizer::detail;
+
+using GraphIdx = size_t;
+struct Edge {
+    Idx2D pos{};
+    Idx weight{};
+};
+
+// TODO(mgovers): investigate whether this really is the correct graph structure
+using TransformerGraph = boost::compressed_sparse_row_graph<boost::directedS, boost::no_property, Edge,
+                                                            boost::no_property, GraphIdx, GraphIdx>;
+
+template <main_core::main_model_state_c State>
+inline auto build_transformer_graph(State const& /*state*/) -> TransformerGraph {
+    // TODO(mgovers): implement
+    return {};
+}
+
+inline auto determine_distance_from_source(TransformerGraph const& /*graph*/) -> std::vector<std::pair<Idx2D, Idx>> {
```

Comment:
```suggestion
inline auto get_distances_from_source(TransformerGraph const& /*graph*/) -> std::vector<std::pair<Idx2D, EdgeWeight >> {
```

Reply:
The more I read the OneNote, the more I feel that it's not `distances`, but `edge_weights`?